RFC.md

```diff
@@ -547,6 +552,21 @@ The clients do the following when trying to establish a trust relationship to an
    2. If no: Show a warning that initiating an encrypted share is not possible to the user.
 5. Store the user certificate locally for next operations
 
+If certificates are externally managed, the workflow will differ:
+
+1. Check if a certificate for the specified user ID is already downloaded (Trust On First Use (TOFU))
+   1. If a certificate is available, it is verified to still be valid.
+      1. It is valid, it will be used.
+      2. It is no longer valid, the client will continue at 2.
+   2. If none is available, the client will continue at 2.
+2. Query the user certificate by sending GET request to the `/ocs/v2.php/apps/end_to_end_encryption/api/v1/public-key` endpoint and sending a JSON encoded `users` parameter containing the specified UIDs
+3. If the user has not yet set up E2E and thus no public key, it will not be possible to share with the user.
+   A warning should then be shown.
+4. Verify that the certificate is valid.
```

Comment:
How is this done? Against which public key?

Reply:
usual validation rules for a x509 certificate (like what a browser would do)
(start valid date, end valid date, not self-signed, not revoked, signing certificates known and valid, ...)

Reply:
we generally are unable to do this for certificates generated by the Nextcloud embedded CA
to check if we could modify the behavior to always use standard certification validation code (again what a browser would check)

Reply:
> signing certificates known and valid

I was concerned about this step. Checking that the x509 certificate is signed by the public key of the NC instance is easy. But who is signing the certificate in this case? A publicly known CA? When you say *"like what a browser would do"*, you mean having a big list of trusted CA, and validating the certificate against them?

Reply:
yes
to do that I needed to fetch intermediate CA certificates from the windows store and let them known to Qt SSL API for validation purposes
I highly doubt that is currently relevant for web end-to-end encryption. I would expect a dedicated development to be needed to fetch the certificate from an user store instead of from the Nextcloud server app